extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/TaskLaneCapacityPolicy.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.k8s.overlord.common;
+
+/**
+ * Enumerates the policies for managing task lane capacities in a task scheduling environment.
+ * Task lanes are logical divisions of task execution resources, allowing for more granular control over how tasks are allocated and executed.
+ * This enumeration defines the strategies that can be applied to these lanes in terms of their capacity management.
+ */
+public enum TaskLaneCapacityPolicy
```

Comment:
I feel these changes should go in core druid. 
Also the AutoCompaction system should use the same mechanisms. No ?